### PR TITLE
Don't force create handle in TabPageCollectionTests.Clear

### LINKED ISSUE #2275: TabPageCollection.Clear forces Handle creation
```cs
        protected void RemoveAll()
        {
            Controls.Clear();

            SendMessage(NativeMethods.TCM_DELETEALLITEMS, 0, 0);
            _tabPages = null;
            _tabPageCount = 0;
        }
```

### PR DESCRIPTION
**Pull request by hughbe**
## Proposed Changes
- Don't force create handle in TabPageCollectionTests.Clear

Fixes #2275

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2519)

### CODE REVIEW

**Comment by hughbe** [comment]
@sharwell hey is there a way to prevent test code from appearing in the `diff` for coverlet? I'm asking because the diff should really only show product code but currently includes test code (e.g. https://codecov.io/gh/dotnet/winforms/pull/2519/diff#D2-1339)

**Comment by RussKie** [comment]
I think this may be because of these lines:
https://github.com/dotnet/winforms/blob/1dc4d5d37c906743aa91e1907f7e69226c0e575e/codecov.yml#L29-L35

https://docs.codecov.io/docs/flags

¯\_(ツ)_/¯

**Comment by hughbe** [comment]
Might be why. Good to merge, though?

File: src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs [hunk 1]
```diff
@@ -1666,7 +1667,11 @@ protected void RemoveAll()
         {
             Controls.Clear();
 
-            SendMessage((int)ComCtl32.TCM.DELETEALLITEMS, 0, 0);
+            if (IsHandleCreated)
+            {
+                User32.SendMessageW(this, ((User32.WindowMessage)TCM.DELETEALLITEMS), IntPtr.Zero, IntPtr.Zero);
```
Comment (RussKie) [inline src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: I think we have an overload
```suggestion
                User32.SendMessageW(this, ((User32.WindowMessage)TCM.DELETEALLITEMS));
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControl.TabPageCollectionTests.cs [hunk 2]
```diff
@@ -1361,6 +1444,23 @@ void parentHandler(object sender, LayoutEventArgs e)
...
+            using var child2 = new TabPage();
+            using var child3 = new TabPage();
+            var collection = new TabControl.TabPageCollection(owner);
+            collection.Add(child1);
+            collection.Add(child2);
+            collection.Add(child3);
+
+            Assert.NotEqual(IntPtr.Zero, owner.Handle);
+            collection.Clear();
+            Assert.Equal((IntPtr)0, User32.SendMessageW(owner.Handle, (User32.WindowMessage)ComCtl32.TCM.GETITEMCOUNT, IntPtr.Zero, IntPtr.Zero));
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControl.TabPageCollectionTests.cs]:
```suggestion
            Assert.Equal((IntPtr)0, User32.SendMessageW(owner.Handle, (User32.WindowMessage)ComCtl32.TCM.GETITEMCOUNT));
```

Comment (hughbe) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControl.TabPageCollectionTests.cs]: This overload doesn't exist yet! do you want me to create? will conflict with existing PRs!

File: src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs (hunk 1 above)
Comment (hughbe) [reply src/System.Windows.Forms/src/System/Windows/Forms/TabControl.cs]: nope sadly

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControl.TabPageCollectionTests.cs (hunk 2 above)
Comment (RussKie) [reply src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/TabControl.TabPageCollectionTests.cs]: No worries, must be thinking of some other method.

**Pull request merged**